Lint

kklin opened the pull request.

kklin commented:
Currently still testing the travis hook, will ping when ready.

kklin commented:
OK @kayousterhout, ready for review

kayousterhout commented on package.json:
```diff
@@ -3,7 +3,15 @@
     "@quilt/quilt": "quilt/quilt"
+  },
+  "scripts": {
+    "lint": "eslint .",
```
Can you add " || true" here and below? That way when lint fails you don't get this ugly npm error:

```
npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "lint"
npm ERR! node v6.10.3
npm ERR! npm  v4.2.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @quilt/nginx@0.0.1 lint: `eslint .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @quilt/nginx@0.0.1 lint script 'eslint .'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @quilt/nginx package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint .
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @quilt/nginx
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @quilt/nginx
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/kayousterhout/.npm/_logs/2017-07-27T19_38_03_438Z-debug.log
```

kklin commented:
@kayousterhout pushed several updates:
- Switched to Gulp for running the lint checks
- Switched to AirBnb's style guide (this required a couple more changes to comply than Google's style guide)
- Added the Quilt compilation Travis check

https://github.com/quilt/quilt/issues/1014 should be reviewed as well

kayousterhout commented on gulpfile.js:
```diff
@@ -0,0 +1,30 @@
+  gulp.src(lintFiles)
+    .pipe(eslint({ fix: true }))
+    .pipe(eslint.format()) // Output the lint results to the console.
+    .pipe(gulpIf(isFixed, gulp.dest('.'))) // Write the fixed files.
```
Why bother with this check? (If nothing was fixed it seems harmless to re-write the file)

kklin replied:
It seems weird to me to always re-write the file. It's unnecessary disk operations, and I wonder if it'd screw with file attributes like "last modified"

kayousterhout replied:
Yup it would definitely update last modified

kayousterhout commented on gulpfile.js:
```diff
@@ -0,0 +1,30 @@
+const gulpIf = require('gulp-if');
+const eslint = require('gulp-eslint');
+
+const lintFiles = ['**/*.js', '!node_modules/**'];
```
Can you add to the commit message how to run the linting?

kayousterhout commented on spark.js:
```diff
@@ -36,7 +36,7 @@ function Spark(nMaster, nWorker, zookeeper) {
     }
 
-    this.public = function() {
+    this.allowFromPublic = function() {
```
As long as you're changing this, can you change this to a more helpful name -- like "exposeUIToPublic"?  (Otherwise it's not obvious what's being allowed)

kayousterhout commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+    - npm install -g gulp-cli
+
+script:
+    - gulp
```
Feel free to ignore -- but I think this would be clearer if you explicitly used "gulp lint" (otherwise if someone wants to know what this is doing, they have to go read the gulp file)

kklin commented:
@kayousterhout Ready for more review

kklin commented:
@kayousterhout does https://github.com/quilt/quilt/pull/1014 look good as well?
